platform/android/app/src/main/kotlin/coop/polypoly/polypod/features/FeatureStorage.kt

```diff
@@ -86,47 +43,36 @@ object FeatureStorage {
         return null
     }
 
-    private fun importFeature(context: Context, id: String) {
-        logger.debug("Installing $id from assets")
-        val source = context.assets.open("features/$id.zip")
+    private fun copyFeatureCategories(context: Context) {
+        val source = context.assets.open("features/categories.json")
         val featuresDir = getFeaturesDir(context)
-        featuresDir.mkdirs()
-        val destination = FileOutputStream(File(featuresDir, "$id.zip"))
+        val destination = FileOutputStream(File(featuresDir, "categories.json"))
         source.copyTo(destination)
     }
 
-    private fun readCategories(context: Context): List<RawCategory> {
-        val categoriesJson = context.assets
-            .open("features/categories.json")
-            .reader()
-            .readText()
-        return Gson()
-            .fromJson(categoriesJson, Array<RawCategory>::class.java)
-            .toList()
-    }
-
-    private fun loadFeature(context: Context, fileName: String): Feature {
-        val content = ZipFile(File(getFeaturesDir(context), "$fileName.zip"))
-        val manifest = readManifest(content)
-        return Feature(fileName, content, context, manifest)
-    }
-
-    private fun readManifest(
-        content: ZipFile
-    ): FeatureManifest? {
-        val manifestEntry = content.getEntry("manifest.json")
-        if (manifestEntry == null) {
-            logger.warn("Missing manifest for '${content.name}'")
-            return null
-        }
-        val manifestString =
-            content.getInputStream(manifestEntry).reader().readText()
+    private fun copyFeatures(context: Context) {
+        val features = context
+            .assets
+            .list("features")
+            ?.filter {
+                it.endsWith("zip")
+            }
 
-        return try {
-            Core.parseFeatureManifest(manifestString)
-        } catch (ex: Exception) {
-            logger.error(ex.message)
-            null
+        if (features == null || features.isEmpty()) {
+            logger.error("No feature were found")
```

Comment:
Just logged? No exception thrown?

Reply:
I wanted to keep the old functionality that just logs the error if features are missing. But I think I would go and crash the app in this case, since if features are missing it is a packaging error(more like developer error)